### PR TITLE
Use MIMEType for image download extension if not present in resource URL

### PR DESCRIPTION
**Pull request by akivab**
Recommend adding so that we can be creative about extension (without forcing the user to explicitly set in the URL provided, as it might be provided in storage metadata)

Addresses the following issue
https://github.com/firebase/firebase-ios-sdk/issues/6590

### CODE REVIEW

**Comment by akivab** [comment]
@chliangGoogle I could move some of this logic to another function to add tests, let me know what your preference would be. Thanks!

**Comment by paulb777** [comment]
@akivab Thanks for the issue and PR! Routing to @chliangGoogle for review.

**Comment by akivab** [comment]
I have tried adding some tests but having trouble with automatic signing issues for the AppHost-FirebaseMessaging-Unit-Tests target (`ld: file not found: /Users/akivab/Library/Developer/Xcode/DerivedData/FirebaseMessaging-hkzirhinkydyvnexsqhhikscqmru/Build/Products/Debug-maccatalyst/AppHost-FirebaseMessaging-Unit-Tests.app/Contents/MacOS/AppHost-FirebaseMessaging-Unit-Tests`)

Attaching the diff so that it might serve as an example of the tests I'm planning to add (but that I didn't want to add due to the concern of being unable to run locally)
[changes_with_tests.txt](https://github.com/firebase/firebase-ios-sdk/files/5296104/changes_with_tests.txt)

**Comment by paulb777** [comment]
It might be easier to test on iOS than Catalyst. You can generate an iOS dev environment with:

`pod gen FirebaseMessaging.podspec --auto-open --local-sources=./ --platforms=ios`

**Comment by akivab** [comment]
Thanks so much @paulb777 -- I've updated my PR with some simple unit tests. @chliangGoogle open to any feedback.

File: FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m
Comment (akivab) [inline FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m:66]: I chose ".tmp" here because I wanted it to fail if there was no storage metadata included and no extension in the filename, but on reflection may make sense to default to ".png" which is allows us to at the very least make a simple assumption on the user's behalf rather than simply fail. Thoughts?

File: FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m
Comment (chliangGoogle) [inline FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m:69]: maybe just 
if (suggestedPathExtension.length > 0) {

File: FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m
Comment (chliangGoogle) [inline FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m:66]: Can you move both static string on top of this file following the other static parameters?

File: FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m
Comment (chliangGoogle) [inline FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m:75]: maybe just replace with
return [response.MIMEType stringByReplacingOccurrencesOfString:kImagePathPrefix withString:@"."]

**Review by chliangGoogle — CHANGES_REQUESTED** [review]
Awesome. Thank you for doing this!

File: FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m
Comment (chliangGoogle) [inline FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m:77]: I don't know image converting enough to know that if it's okay to return a default format like "png" if it is not a png image?
maybe if neither format is valid, we should return empty string and let it fail with the error before?
I don't know if it's also secure to download file that might not be image at all. wdyt?

Comment (akivab) [reply FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m:77]: I figured we'd want to set to ".png" in case the metadata wasn't stored, but agreed we can let it error instead.

**Comment by akivab** [comment]
Thanks @chliangGoogle, addressed comments.

File: FirebaseMessaging/Tests/UnitTests/FIRMessagingExtensionHelperTest.m
Comment (chliangGoogle) [inline FirebaseMessaging/Tests/UnitTests/FIRMessagingExtensionHelperTest.m:53]: make sure to call stopMocking in tearDown

**Review by chliangGoogle — CHANGES_REQUESTED** [review]
one last nit bit.

Can you also update the changelog in FirebaseMessaging/CHANGELOG?
#unreleased 
[fixed] Fixed the issue that failed to download image if no extension but MIME type is set. (#6590)
- [fixed] Fix issue :3

File: FirebaseMessaging/CHANGELOG.md
Comment (chliangGoogle) [inline FirebaseMessaging/CHANGELOG.md:3]: Fixed an issue that downloading an image failed when there's no extension in the file name but MIME type is set.

**Comment by akivab** [comment]
Thanks @chliangGoogle  -- added to the CHANGELOG.md and calling teardown on the mock now

**Review by chliangGoogle — COMMENTED** [review]
Sorry, my wording is bad, I corrected mine again. 

Thank you very much doing this. I will approve this when it's okay to merge master. Our team is doing maintenance work at the moment. Once it's cleared, I will approve and you can merge to master.

File: FirebaseMessaging/CHANGELOG.md
Comment (akivab) [reply FirebaseMessaging/CHANGELOG.md:3]: Sounds good, updated language.

**Comment by akivab** [comment]
Perfect, sounds good. Contributing here has been a great experience, hoping to do more in the future.

**Review by chliangGoogle — APPROVED** [review]

**Comment by akivab** [comment]
Hi @chliangGoogle, I don't have write access to this repo so expecting that you might be able to merge in. Thanks so much.

**Comment by paulb777** [comment]
I'll merge it now - to make sure that none of the upcoming major changes introduce a conflict.

Thanks for the PR @akivab !

**Pull request merged**
